common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDB.java

```diff
@@ -176,11 +176,13 @@ public void writeAll(List<?> values) throws Exception {
       final Iterator<byte[]> serializedValueIter;
 
       // Deserialize outside synchronized block
-      List<byte[]> list = new ArrayList<>(entry.getValue().size());
-      for (Object value : values) {
-        list.add(serializer.serialize(value));
+      {
```

Comment:
Probably I'm not aware of Java spec; we are just adding a scope here and everything remains the same. What is changed? Sorry for ignorant question.

Reply:
You are right, scope change is not strictly required :)
The main change is the `values` -> `entry.getValue()`.

Reply:
Ah OK, probably better to not explicitly scope as it's still clear without highlighting that code block.

Btw, looks like we are performing whole write "multiple times" if there are multiple types? Wow. Nice finding!

> Typical usecase is for this to be a single class.

This is the thing, OK.

Reply:
Updated.